### PR TITLE
gRFC for static method and host.

### PR DESCRIPTION
**Pull request by soheilhy**
This proposal requires a major version bump.

### CODE REVIEW

**Comment by soheilhy** [comment]
Thread on grpc-io https://groups.google.com/forum/#!topic/grpc-io/iYwUVQKU-OU

File: L52-core-static-method-host.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## Background
+
+For pre-registering, `method` and `host` are pre-registered on channels using
+`grpc_channel_register_call()`. Although no user that we are aware of uses
+non-static strings for `method` and `host`, we have to conservatively intern the
+strings since `grpc_channel_register_call()` is not clear about the ownership
+and life-time of `method` and `host`. Interning `method` and `host` results in
+unnecessary ref-counting and cache-line contention for every single call. This
+can be eliminated by making static.
```
Comment (chwarr) [inline L52-core-static-method-host.md]: Minor: I think there's a "them" missing in the last sentence. "by making _them_ static."

File: L52-core-static-method-host.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Static strings for `method` and `host` for the input to `grpc_channel_register_call()`
+----
+* Author(s): [Soheil Hassas Yeganeh](https://github.com/soheilhy)
+* Approver: vjpai, yashykt
+* Status: In Review
+* Implemented in: core
+* Last updated: 2019-06-05
+* Discussion at: <google group thread> (filled after thread exists)
```
Comment (chwarr) [inline L52-core-static-method-host.md]: Minor: update with https://groups.google.com/forum/#!topic/grpc-io/iYwUVQKU-OU

File: L52-core-static-method-host.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Proposal
+
+Require static string for the `method` and `host` arguments to
+`grpc_channel_register_call()`. That is the `method` and `host` strings must
+be kept valid while the server is up.
+
+## Rationale
+
+`method` and `host` are reffed for every call. Doing an atomic fetch-add per
+call is quite expensive, and truly unnecessary since the strings can be easily
```
Comment (chwarr) [inline L52-core-static-method-host.md]: Minor: consider "can easily be kept alive"

File: L52-core-static-method-host.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+kept alive while the channel is alive.
+
+## Implementation
+
+This is basically a simple comment update and removing `grpc_slice_intern()`
+calls for `mehtod` and `host`, as implemented in
```
Comment (chwarr) [inline L52-core-static-method-host.md]: Minor: "method"

Comment (soheilhy) [reply L52-core-static-method-host.md]: Done.

File: L52-core-static-method-host.md (hunk 3 above)
Comment (soheilhy) [reply L52-core-static-method-host.md]: Done.

File: L52-core-static-method-host.md (hunk 2 above)
Comment (soheilhy) [reply L52-core-static-method-host.md]: Done.

File: L52-core-static-method-host.md (hunk 1 above)
Comment (soheilhy) [reply L52-core-static-method-host.md]: Done.

**Comment by soheilhy** [comment]
Thank you for the review @chwarr

**Comment by soheilhy** [comment]
Thanks!

**Pull request merged**